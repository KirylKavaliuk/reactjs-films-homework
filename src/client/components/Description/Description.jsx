import React from 'react';
import PropTypes from 'prop-types';

import { withConditionalRendering } from 'utils/rendering';

import classNames from 'classnames';

import styles from './Description.scss';

const Description = ({ open, text }) => (
  <div className={
    classNames(
      styles.description,
      { [styles.open]: open },
    ) }>
    { text }
  </div>
);

Description.defaultProps = {
  open: false,
  text: '',
};

Description.propTypes = {
  open: PropTypes.bool,
  text: PropTypes.string,
};

export default withConditionalRendering(Description, 'text');
