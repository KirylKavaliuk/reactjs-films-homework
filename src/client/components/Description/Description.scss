@import 'styles/palette';

.description {
  position: absolute;
  top: -215px;
  right: 0;
  overflow: auto;
  box-sizing: border-box;
  width: 300px;
  max-height: 180px;
  padding: 13px;
  background: rgba(0, 0, 0, 0.75);
  line-height: 2;

  &::-webkit-scrollbar {
    display: none;
  }
}

@media only screen and (max-width: 450px) {
  .description {
    top: -190px;
  }
}
