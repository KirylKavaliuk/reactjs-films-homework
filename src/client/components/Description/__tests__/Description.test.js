import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Description from '../Description';

it('renders description open shallow correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Description open text='text'/>);

  const tree = renderer.getRenderOutput();

  expect(tree).toMatchSnapshot();
});

it('renders description close shallow correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Description/>);

  const tree = renderer.getRenderOutput();

  expect(tree).toMatchSnapshot();
});
