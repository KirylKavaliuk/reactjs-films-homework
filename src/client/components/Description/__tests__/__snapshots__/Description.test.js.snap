// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders description open shallow correctly 1`] = `
<Description
  open={true}
  text="text"
/>
`;
